### PR TITLE
feat: support depth URL argument in Terragrunt modules

### PR DESCRIPTION
**Pull request by emil2k**
## Changes

Add Support for renovating Terragrunt Modules from Git repositories using shallow clone URL argument by expanding the Regex

## Context

Similar to https://github.com/renovatebot/renovate/pull/21287 but for Terragrunt.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/terragrunt/extract.spec.ts [hunk 1]
```diff
@@ -14,6 +14,20 @@ describe('modules/manager/terragrunt/extract', () => {
       expect(res?.deps.filter((dep) => dep.skipReason)).toHaveLength(5);
     });
 
+    it('extracts terragrunt sources with depth specified after the branch', () => {
+      const res = extractPackageFile(Fixtures?.get('3.hcl'));
+      expect(res).toMatchSnapshot();
```
Comment (secustor) [inline lib/modules/manager/terragrunt/extract.spec.ts:19]: Do not use `toMatchSnapshot` rather use `toEqual`

We have newer code style guides https://github.com/renovatebot/renovate/blob/main/docs/development/best-practices.md#unit-testing

File: lib/modules/manager/terragrunt/extract.spec.ts [hunk 2]
```diff
@@ -14,6 +14,20 @@ describe('modules/manager/terragrunt/extract', () => {
...
+      expect(res?.deps).toHaveLength(30);
+      expect(res?.deps.filter((dep) => dep.skipReason)).toHaveLength(5);
+    });
+
+    it('extracts terragrunt sources with depth specified before the branch', () => {
+      const res = extractPackageFile(Fixtures?.get('4.hcl'));
+      expect(res).toMatchSnapshot();
```
Comment (secustor) [inline lib/modules/manager/terragrunt/extract.spec.ts:26]: Do not use `toMatchSnapshot` rather use `toEqual`

File: lib/modules/manager/terragrunt/extract.spec.ts [hunk 3]
```diff
@@ -14,6 +14,20 @@ describe('modules/manager/terragrunt/extract', () => {
...
 
+    it('extracts terragrunt sources with depth specified after the branch', () => {
+      const res = extractPackageFile(Fixtures?.get('3.hcl'));
+      expect(res).toMatchSnapshot();
+      expect(res?.deps).toHaveLength(30);
+      expect(res?.deps.filter((dep) => dep.skipReason)).toHaveLength(5);
+    });
+
+    it('extracts terragrunt sources with depth specified before the branch', () => {
+      const res = extractPackageFile(Fixtures?.get('4.hcl'));
```
Comment (secustor) [inline lib/modules/manager/terragrunt/extract.spec.ts:25]: I don't think the optional chaining is necessary.

**Review by secustor — CHANGES_REQUESTED** [review]

File: lib/modules/manager/terragrunt/__fixtures__/4.hcl [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+terraform {
+  source = "my.host"
+}
+
+#local hostname
+terraform {
+  source = "my.host.local/sources/example?depth=1&ref=v1.2.1"
+}
+
+#ip
```
Comment (secustor) [inline lib/modules/manager/terragrunt/__fixtures__/4.hcl:66]: This is not Ip.

File: lib/modules/manager/terragrunt/modules.ts [hunk 5]
```diff
@@ -8,10 +8,10 @@ import { extractTerragruntProvider } from './providers';
 import type { ExtractionResult, TerraformManagerData } from './types';
 
 export const githubRefMatchRegex = regEx(
-  /github\.com([/:])(?<project>[^/]+\/[a-z0-9-_.]+).*\?ref=(?<tag>.*)$/i
+  /github\.com([/:])(?<project>[^/]+\/[a-z0-9-_.]+).*\?(depth=\d+&)?ref=(?<tag>.*?)(&depth=\d+)?$$/i
```
Comment (secustor) [inline lib/modules/manager/terragrunt/modules.ts:11]:
```suggestion
  /github\.com([/:])(?<project>[^/]+\/[a-z0-9-_.]+).*\?(depth=\d+&)?ref=(?<tag>.*?)(&depth=\d+)?$/i
```
Why the double dollar sign?

File: lib/modules/manager/terragrunt/modules.ts [hunk 6]
```diff
@@ -8,10 +8,10 @@ import { extractTerragruntProvider } from './providers';
...
 );
 export const gitTagsRefMatchRegex = regEx(
-  /(?:git::)?(?<url>(?:http|https|ssh):\/\/(?:.*@)?(?<path>.*.*\/(?<project>.*\/.*)))\?ref=(?<tag>.*)$/
+  /(?:git::)?(?<url>(?:http|https|ssh):\/\/(?:.*@)?(?<path>.*.*\/(?<project>.*\/.*)))\?(depth=\d+&)?ref=(?<tag>.*?)(&depth=\d+)?$$/
```
Comment (secustor) [inline lib/modules/manager/terragrunt/modules.ts:14]: Same here.

File: lib/modules/manager/terragrunt/__fixtures__/4.hcl [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+#bar
+terraform {
+  source = "github.com/hashicorp/example?depth=1&ref=next"
+}
+
+#hostname
+terraform {
+  source = "https://104.196.242.174"example?depth=1&ref=next"
+}
```
[the remark this answers is not in the document]
Comment (emil2k) [reply lib/modules/manager/terragrunt/__fixtures__/4.hcl:39]: This is basically taken from the preexisting fixture, but I could make the change.

**Review by viceice — COMMENTED** [review]
Do we really need so many fixtures?

**Review by secustor — APPROVED** [review]

**Pull request merged**